schema_based_on_diagram.sql

```diff
@@ -0,0 +1,43 @@
+CREATE TABLE patients (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255),
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id),
+  status VARCHAR (50) 
+);
+
+ CREATE TABLE treatments (
+  id SERIAL PRIMARY KEY,
+  type VARCHAR (50),
+  name VARCHAR (40)
+);
+
+CREATE TABLE medical_history_treatments (
+    id SERIAL PRIMARY KEY,
+    medical_history_id INT REFERENCES medical_histories(id),
+    treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE TABLE invoices (
+  id SERIAL PRIMARY KEY,
+  totat_amount DECIMAL(10,2),
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT UNIQUE REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL(10,2),
+  quantity INT,
+  invoice_id INT REFERENCES invoices(id),
+  treatment_id INT REFERENCES treatments(id)
+);
```

Comment:
- [ ] Kindly create indexes on the foreign keys you created in this schema file. Creating indexes on foreign keys makes queries more efficient especially if you are querying a table that has many data entries. 👍